.github/workflows/ci-test.yml

```diff
@@ -238,59 +238,25 @@ jobs:
           fi
           echo "::set-output name=prev-tag::$(echo $res)";
 
-  land-blocking-switch:
-    runs-on: ubuntu-latest
-    if: ${{ always() }}
-    needs:
-      - build-images
-      - need-base-images
-    outputs:
-      lbt-switch: ${{ steps.set-switch.outputs.lbt-switch }}
-      head-tag: ${{ steps.set-switch.outputs.head-tag }}
-      prev-tag: ${{ steps.set-switch.outputs.prev-tag }}
-    steps:
-      - name: set lbt switch value
-        id: set-switch
-        run: |
-          lbt=true
-          head=''
-          prev=''
-          if ${{ secrets.KILL_SWITCH_LAND_BLOCKING_TEST || needs.build-images.result!='success' }}; then
-            lbt=false
-          elif ${{ needs.need-base-images.result=='skipped' }}; then
-            head=${{ needs.build-images.outputs.head-tag }}
-          elif ${{ needs.need-base-images.result=='success' }}; then
-            head=${{ needs.build-images.outputs.head-tag }}
-            prev=${{ needs.need-base-images.outputs.prev-tag }}
-          else
-            lbt=false
-          fi
-          echo "::set-output name=lbt-switch::$(echo $lbt)";
-          echo "::set-output name=head-tag::$(echo $head)";
-          echo "::set-output name=prev-tag::$(echo $prev)";
-
   land-blocking-test:
     name: Run land blocking test
     runs-on: self-hosted
-    needs: [prepare, land-blocking-switch]
-    if: ${{ always() }}
+    needs: [prepare, build-images, need-base-images]
+    if: ${{ always() && needs.build-images.result=='success' }}
     timeout-minutes: 30
     steps:
       - name: Launch cluster test
-        if: ${{ needs.land-blocking-switch.outputs.lbt-switch == 'true' }}
         # NOTE Remember to update PR comment payload if cti cmd is updated.
         run: |
           set +e
           date
           export CTI_OUTPUT_LOG=$(mktemp)
           echo "CTI_OUTPUT_LOG=$CTI_OUTPUT_LOG" >> $GITHUB_ENV
           cmd=""
-          if [ "${{ needs.land-blocking-switch.outputs.prev-tag }} == ''" ]; then
-            echo here
-            cmd="./scripts/cti --tag ${{ needs.land-blocking-switch.outputs.head-tag }} -E RUST_LOG=debug --report report.json --suite land_blocking"
+          if ${{ needs.need-base-images.result!='success' }}; then
+            cmd="./scripts/cti --tag ${{ needs.build-images.outputs.head-tag }} -E RUST_LOG=debug --report report.json --suite land_blocking"
```

Comment:
uh, based on 245, and 256, then 257 should be unreachable.   I think there is a flaw in this reasoning.

Reply:
scratch that.  I mis read, carry on.